### PR TITLE
Fix scraper to use webfeeds for distribution area

### PR DESCRIPTION
**Pull request by danielleung27**
Fix scraper to use webfeeds for distribution area

### CODE REVIEW

**Review by bnehoran — DISMISSED** [review]

File: tigerpath/scraper/scrape_parse.py [hunk 1]
```diff
@@ -32,21 +141,27 @@ def scrape_parse_semester(term_code):
...
+            ParseError("key " + key + " does not exist")
+        else:
+            return h.unescape(found.text)
 
     def get_current_semester():
         """ get semester according to TERM_CODE
         """
         #global CURRENT_SEMESTER
+        # for now hardwire the namespaces--too annoying
+        PTON_NAMESPACE = u'http://as.oit.princeton.edu/xml/courseofferings-1_4'
```
Comment (danielleung27) [inline tigerpath/scraper/scrape_parse.py:164]: Why don't we just use namespace 1_5, does it break this function?

File: tigerpath/scraper/scrape_parse.py [hunk 2]
```diff
@@ -220,4 +319,4 @@ def remove_namespace(doc, namespace):
             if elem.tag.startswith(ns):
                 elem.tag = elem.tag[nsl:]
 
-    return scrape_all()
```
Comment (danielleung27) [inline tigerpath/scraper/scrape_parse.py:223]: get rid of this

**Review by bnehoran — APPROVED** [review]

File: tigerpath/scraper/scrape_parse.py [hunk 3]
```diff
@@ -32,21 +141,27 @@ def scrape_parse_semester(term_code):
...
 
     CURRENT_SEMESTER = ['']
 
     h = HTMLParser()
 
-    def get_text(key, object):
-        return h.unescape(raise_if_none(object.find(key), "key " + key + " does not exist").text)
+    def get_text(key, object, fail_ok=False):
+        found = object.find(key)
+        if fail_ok and (found is None or found.text is None):
```
Comment (danielleung27) [inline tigerpath/scraper/scrape_parse.py:152]: I think it's more readable like this
`if fail_ok:`
____`if(found is None or found.text is None):`
________`return found`
____`else:`
________`return h.unescape(found.text)`
`else:`
 ____`return ParseError("key " + key + " does not exist")`

File: tigerpath/scraper/scrape_parse.py (hunk 1 above)
Comment (bnehoran) [reply tigerpath/scraper/scrape_parse.py:164]: Yeah, updating to 1.5 breaks it. Not sure why.

File: tigerpath/scraper/scrape_parse.py (hunk 3 above)
Comment (bnehoran) [reply tigerpath/scraper/scrape_parse.py:152]: That doesn't work in the case of `fail_ok=False`

File: tigerpath/scraper/scrape_parse.py (hunk 2 above)
Comment (bnehoran) [reply tigerpath/scraper/scrape_parse.py:223]: Shouldn't there be a newline at the end of python files? I thought that was the standard.
I know it's not related to the PR, but also it doesn't hurt.

Comment (danielleung27) [reply tigerpath/scraper/scrape_parse.py:223]: oh ok, thats fine then. i thought it was an accident

File: tigerpath/scraper/scrape_parse.py (hunk 3 above)
Comment (danielleung27) [reply tigerpath/scraper/scrape_parse.py:152]: oh wait i see, my bad.

**Pull request merged**